pywbem_mock/_namespaceprovider.py

```diff
@@ -0,0 +1,290 @@
+                        new_instance.classname))
+
+        # Normalize the namespace name
+        # TODO: Somehow we need to get this out of the way.
+        new_namespace = new_namespace.strip('/')
+
+        # Write it back to the instance in case it was changed
+        new_instance['Name'] = new_namespace
+
+        # These values must match those in
+        # tests/unittest/utils/wbemserver_mock.py.
+        try:
+            if new_instance['CreationClassName'] != new_instance.classname \
+                or new_instance['ObjectManagerName'] != OBJECTMANAGERNAME \
+                or new_instance['ObjectManagerCreationClassName'] != \
+                'CIM_ObjectManager' \
+                or new_instance['SystemName'] != SYSTEMNAME \
+                or new_instance['SystemCreationClassName'] != \
+                    SYSTEMCREATIONCLASSNAME:
+
+                raise CIMError(
+                    CIM_ERR_INVALID_PARAMETER,
+                    _format("Namespace creation via CreateInstance: "
+                            "Invalid property for CIM_Instance {0!A}",
+                            new_instance))
+
+        except KeyError as ke:
+            raise CIMError(
+                CIM_ERR_INVALID_PARAMETER,
+                _format("Namespace creation via CreateInstance: "
+                        "Missing property for CIM_Instance {0!A}. Exception "
+                        "{1!A}", new_instance, ke))
+
+        if new_namespace not in self.cimrepository.namespaces:
+            self.cimrepository.add_namespace(new_namespace)
+
+        # Pass NewInstance to the default CreateInstance
+        # Default CreateInstance creates path, validates all key properties
+        # in instance and inserts instance into the repository.
+        return super(CIMNamespaceProvider, self).CreateInstance(namespace,
+                                                                new_instance)
+
+    def ModifyInstance(self, ModifiedInstance,
+                       IncludeQualifiers=None, PropertyList=None):
+        """
+        Modification of CIM_Namespace instance not allowed
+        """
+        raise CIMError(
+            CIM_ERR_NOT_SUPPORTED,
+            _format("Modification of CIM_Namespace {0!A} not implemented. ",
+                    ModifiedInstance.classname))
+
+    def DeleteInstance(self, InstanceName):
+        """
+        Delete the namespace defined by the Name property in InstanceName
+        and remove the instance defined by InstanceName
+        """
+        remove_namespace = InstanceName.keybindings['Name']
+
+        # Reflect the namespace deletion in the CIM repository
+        # This will check the namespace for being empty.
+        self.remove_namespace(remove_namespace)
+
+        # Delete the instance from the CIM repository
+        instance_store = self.get_instance_store(InstanceName.namespace)
+        instance_store.delete(InstanceName)
```

Comment:
Deleting the namespace from the repository before deleting the CIM_Namespace instance from the interop namespace implies that this method cannot delete the interop namespace. Which probably makes sense. If we agree that that makes sense, it should be described in the docstring of this method, and there should be a check for that that surfaces a proper CIMError.

Reply:
DONE.  I agree that this provider should not be able to remove the interop namespace so added test and exception.  I also added more to docstring for the DeleteInstance